Sources/Segment/Utilities/Networking/HTTPSession+Apple.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+#if os(Linux) || os(Windows)
+import FoundationNetworking
+#endif
+
+extension URLSessionDataTask: DataTask {}
+
+// Give the built in `URLSession` conformance to HTTPSession so that it can easily be used
+extension URLSession: HTTPSession {
+  public func uploadTask(with request: URLRequest, fromFile file: URL, completionHandler: @escaping (Data?, URLResponse?,  (any Error)?) -> Void) -> any DataTask {
```

Comment:
I think you can avoid these methods if you define an `associatedtype` for `DataTask` in the protocol definition. See below. Then all you would need to do is declare that `URLSessionDataTask` conforms to `DataTask`.